components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -393,6 +396,10 @@ && isStepHasMultiOption(context)
 
                 if (!context.isLogoutRequest()) {
                     FrameworkUtils.getAuthenticationRequestHandler().handle(request, responseWrapper, context);
+                    // Adding spIp param to the redirect URL if the authenticator is not the organization authenticator.
+                    if (!ORGANIZATION_AUTHENTICATOR.equals(request.getParameter(AUTHENTICATOR))) {
```

Comment:
Here we are trying skip adding sp-uuid to the consent page when it is organization authenticator. Trying to understand the how it created an issue. I hope you can proceed with the improvement exclude this and explain the issue may be via sharing the network trace where we can identify how it caused an issue.

Reply:
this is not for consent page. It's for the MFA pages in sub org applications. 
When organization authenticator is being used, it's the point where the application auth process switches from main org to sub org. At this point the flow is still being executed at the main org level application. But the url being generated here leads to sub org first factor page. If we add the spId in the context to the URL, it'll be the main org application id, which is not valid for the sub org application page.